### ISSUE TITLE
Future style changes

### ISSUE THREAD

**Issue by timhoffm** [matplotlib/matplotlib#14331]
This is a collection to keep track of possibly relevant ideas, which we should consider if we ever do a new style (likely not before 4.0).

- `axhline`, `axvline` should use the grid color by default https://github.com/matplotlib/matplotlib/issues/14222#issuecomment-495964031
- Color blind-friendly default color cycle #9460
- How to handle changes in styles? #13680 (Update seaborn style)

**Comment by WeatherGod** [comment]
I would like to propose reverting the removal of the errorbar caps

**Comment by aoz** [comment]
Not sure if this is the right place for this but had a comment about `vlines`: 

I think vlines *should* use the cycle for colors/linestyles, but it doesn't. `vlines` plots a *set* of lines. If I'm doing multiple calls to it, I'm most likely plotting multiple sets of lines each of which represents a different "property", and thus each set should get its own visual representation. If I want a bunch of lines with the same visual representation, I can do a single call to axhline.

I ended up here after spending a bunch of time trying to debug my code and eventually learning about a somewhat convoluted solution to my problem (which I'm still struggling to make work without delving too much into the internals of matplotlib).

**Comment by jklymak** [comment]
Axvline discussion is better had at #14222. Thanks.

**Comment by aoz** [comment]
> Axvline discussion is better had at #14222. Thanks.

If I understand correctly, `axvline` plots a single vertical line. I don't have a strong argument (or need) for cycling in that case.

One closed issue discussing a similar topic was pointing to this issue as a place to collect feature requests on styles, but if there is a more appropriate issue, I'd be happy to copy the comment there.

**Comment by timhoffm** [comment]
I agree, `vlines` should take part in the color cycle. - It's currently fixed to "C0", so no change for the first call. It's also simple to fix the color to "C0" if needed on subsequent calls,

**Comment by babryce** [comment]
With 1 in 12 males colorblind, or about 4% of the user base impacted it is very important to do whatever is needed to move matplotlib to an accessible set of color cycles, styles, etc. 
https://matplotlib.org/stable/users/prev_whats_new/whats_new_2.2.html#new-style-colorblind-friendly-color-cycle

It isn't really enough to have them available because a large fraction of the population just is unaware of the amount of content that is impossible to read/use. Obviously I switch to other styles always. Marginal aesthetics cannot justify excluding 4+% of users, so I think this should be an essential change even if it has to wait to the 4.0 release as noted.

The best research available should be used in picking the new defaults. https://colorbrewer2.org is a starting point. Of particular difficulty are color based contour/heat maps. It is extremely rare to find these rendered in a colorblind safe manner with the only parties doing so other colorblind persons in my 20+ years in science.
